Application Server unit test failure

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

Segmentation fault in Application Server unit tests https://github.com/TheThingsNetwork/lorawan-stack/runs/2828044779#step:15:41

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Be unlucky

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!-- Please paste terminal output, upload logs (as .txt) or upload screenshots. -->

```
--- FAIL: TestSkipPayloadCrypto (1.10s)
    --- FAIL: TestSkipPayloadCrypto/NoOverride/false (1.03s)
        --- FAIL: TestSkipPayloadCrypto/NoOverride/false/Uplink (1.03s)
            --- FAIL: TestSkipPayloadCrypto/NoOverride/false/Uplink/JoinAccept (1.02s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x1dd0f37]

goroutine 3383 [running]:
testing.tRunner.func1.2(0x1fc2da0, 0x36b0380)
	/opt/hostedtoolcache/go/1.16.5/x64/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc0012b8600)
	/opt/hostedtoolcache/go/1.16.5/x64/src/testing/testing.go:1146 +0x4b6
panic(0x1fc2da0, 0x36b0380)
	/opt/hostedtoolcache/go/1.16.5/x64/src/runtime/panic.go:965 +0x1b9
go.thethings.network/lorawan-stack/v3/pkg/applicationserver_test.TestSkipPayloadCrypto.func4.1.2(0xc0012b8600, 0x0)
	/home/runner/work/lorawan-stack/lorawan-stack/pkg/applicationserver/applicationserver_test.go:2473 +0x257
go.thethings.network/lorawan-stack/v3/pkg/applicationserver_test.TestSkipPayloadCrypto.func4.1.6(0xc0012b8600)
	/home/runner/work/lorawan-stack/lorawan-stack/pkg/applicationserver/applicationserver_test.go:2608 +0x178
testing.tRunner(0xc0012b8600, 0xc0010801c0)
	/opt/hostedtoolcache/go/1.16.5/x64/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.16.5/x64/src/testing/testing.go:1238 +0x2b3
FAIL	go.thethings.network/lorawan-stack/v3/pkg/applicationserver	69.444s
```

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

No segfaults.

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->

3.13.1 dev

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->

Check if this is a flaky test or if this is actually a bug. To the best of my knowledge we have not encountered this elsewhere, and happens quite rarely.

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

No

--- a/pkg/applicationserver/applicationserver_test.go
+++ b/pkg/applicationserver/applicationserver_test.go
@@ -2603,7 +2603,7 @@ func TestSkipPayloadCrypto(t *testing.T) {
 							} else {
 								t.Fatalf("Expected no upstream message but got %v", msg)
 							}
-						case <-time.After(Timeout * 2):
+						case <-time.After(Timeout * 4):
 							if step.AssertUp != nil {
 								step.AssertUp(t, nil)
 							} else {

